x-pack/platform/plugins/shared/streams_app/public/components/data_management/stream_detail_enrichment/processors/grok/use_grok_pattern_suggestion.tsx

```diff
@@ -30,42 +37,88 @@ export function useGrokPatternSuggestion() {
   } = useKibana();
 
   const abortController = useAbortController();
-  return useAsyncFn((params: GrokPatternSuggestionParams | null) => {
-    if (params === null) {
-      return Promise.resolve(undefined); // Reset to initial value
-    }
-    const finishTrackingAndReport = telemetryClient.startTrackingAIGrokSuggestionLatency({
-      name: params.streamName,
-      field: params.fieldName,
-      connector_id: params.connectorId,
-    });
-    return streamsRepositoryClient
-      .fetch('POST /internal/streams/{name}/processing/_suggestions', {
-        signal: abortController.signal,
-        params: {
-          path: { name: params.streamName },
-          body: {
-            field: params.fieldName,
-            connectorId: params.connectorId,
-            samples: params.samples,
+  const processorsWithoutCurrent = useSimulatorSelector((snapshot) =>
+    snapshot.context.processors.slice(0, -1)
+  );
+  const previewDocsFilter = useSimulatorSelector((snapshot) => snapshot.context.previewDocsFilter);
+  const originalSamples = useSimulatorSelector((snapshot) =>
+    selectOriginalPreviewRecords(snapshot.context)
+  );
+
+  return useAsyncFn(
+    async (params: GrokPatternSuggestionParams | null) => {
+      if (params === null) {
+        return Promise.resolve(undefined); // Reset to initial value
+      }
+
+      let samples = originalSamples
+        .map((doc) => doc.document)
+        .map(flattenObjectNestedLast) as FlattenRecord[];
+
+      /**
+       * If there are processors, we run a partial simulation to get the samples.
+       * If there are no processors, we use the original samples previously assigned.
+       */
+      if (processorsWithoutCurrent.length > 0) {
+        const simulation = await simulateProcessing({
+          streamsRepositoryClient,
+          input: {
+            streamName: params.streamName,
+            processors: processorsWithoutCurrent,
+            documents: samples,
           },
-        },
-      })
-      .then(
-        (response) => {
-          finishTrackingAndReport(
-            response.length || 0,
-            response.map(
-              ({ simulationResult }) =>
-                simulationResult.processors_metrics['grok-processor'].parsed_rate
-            )
-          );
-          return response;
-        },
-        (error: Error) => {
-          showErrorToast(notifications, error);
-          throw error;
-        }
-      );
-  });
+        });
+
+        samples = selectPreviewRecords({
+          samples: originalSamples,
+          previewDocsFilter,
+          simulation,
+        } as SimulationContext);
+      }
```

Comment:
This now collects the samples for the grok suggestions logic given the logic described [here](https://github.com/elastic/streams-program/issues/255#issuecomment-2824236921).